src/vm/jitinterface.cpp

```diff
@@ -11623,13 +11623,21 @@ void CEEJitInfo::allocMem (
 
     S_SIZE_T totalSize = S_SIZE_T(codeSize);
 
+    bool roData16bytesAligned = ((flag & CORJIT_ALLOCMEM_FLG_RODATA_16BYTE_ALIGN)!= 0);
     if (roDataSize > 0)
     {
-        totalSize.AlignUp(sizeof(void *));
+        if (roData16bytesAligned)
+        {
+            totalSize.AlignUp(16);
+        }
+        else
+        {
+            totalSize.AlignUp(sizeof(void *));
+        }
         totalSize += roDataSize;
 
 #ifndef _WIN64
-        if (roDataSize >= 8)
+        if ((roDataSize >= 8) && !roData16bytesAligned)
```

Comment:
@jkotas and I discussed this offline and indeed there was a problem. I fixed it and also simplified code. @jkotas PTAL.